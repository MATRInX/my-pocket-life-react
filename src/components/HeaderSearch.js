import React from 'react';
import { history } from '../routers/AppRouter';

class HeaderSearch extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      queryString: ''
    };
  };

  onSubmit = e => {
    e.preventDefault();
    const queryString = this.state.queryString;
    this.setState(() => ({ queryString: '' }));    
    history.push(`/szukaj/${queryString}`);
  };

  onQueryStringChange = e => {
    const queryString = e.target.value;
    this.setState(() => ({ queryString }));
  };

  render() {
    return (
      <form onSubmit={this.onSubmit}>
        <input 
          type="search"
          value={this.state.queryString}
          onChange={this.onQueryStringChange}
          placeholder="Szukaj..."
        />
      </form>
    );
  };
};

export default HeaderSearch;