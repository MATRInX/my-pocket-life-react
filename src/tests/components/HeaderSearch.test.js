import React from 'react';
import { shallow } from 'enzyme';
import HeaderSearch from '../../components/HeaderSearch';

test('should correctly render HeaderSearch', () => {
  const wrapper = shallow(<HeaderSearch />);
  expect(wrapper).toMatchSnapshot();
});

test('should set search text on input change', () => {
  const value = 'Typed value';
  const wrapper = shallow(<HeaderSearch />);
  wrapper.find('input').at(0).simulate('change', {
    target: {
      value
    }
  });
  expect(wrapper.state('queryString')).toBe(value);
});