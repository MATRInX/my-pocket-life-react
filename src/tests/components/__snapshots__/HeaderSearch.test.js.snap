// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render HeaderSearch 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Szukaj..."
    type="search"
    value=""
  />
</form>
`;
